### ISSUE TITLE
OGR: Force using X-format for reading input source

### ISSUE THREAD

**Issue by maphew** [pcjericks/py-gdalogr-cookbook#154]
this would probably involve use of [OGR_SKIP](https://trac.osgeo.org/gdal/wiki/ConfigOptions#OGR_SKIP), which *"This can be useful when a driver tries to open a datasource that it should not recognize, or when several drivers are built-in that can open the same datasets (for example KML, LIBKML datasources)."* ... the scenario that caused me to go hunting for solutions and find this cookbook.

**Comment by ustroetz** [comment]
I am not sure if I understand your question right. 
But maybe this goes in the direction you mean.
You can specify a driver by name. This way if you open a file the driver to read is already defined:
```
import ogr
driver = ogr.GetDriverByName('ESRI Shapefile')
data_source = driver.Open(fn, 0)
```

**Comment by maphew** [comment]
I opened a thread on GIS Stack Exchange for this, http://gis.stackexchange.com/questions/141905/force-ogr-to-use-specific-driver-for-input-format. So far it looks like the file extension plays a role too, that specifying driver by name is not enough.

The thrust of my question/suggestion is that I'm looking for a method to force ogr in python or ogr2ogr command line to use one and only one format driver when attempting to read a file, pushing all other magic guessing aside. I thought I just needed to learn the right combination of options, but I'm beginning to think this might need to be an ogr feature request.
